src/server.js

```diff
@@ -23,12 +30,103 @@ const sendUserError = (err, res) => {
   }
 };
 
+const middleVerification = (req, res, next) => {
+  const sess = req.session;
+  if (!sess.LoginId) {
+    sendUserError("Please log in", res);
+    return;
+  }
+  User.findById(sess.LoginId, (err, user) => {
+    if (err) {
+      sendUserError(err, res);
+      return;
+    }
+    req.user = user;
+    next();
+  });
+};
+
+server.use("/restricted/*", (req, res, next) => {
+  const sess = req.session;
+  if (!sess.LoginId) {
+    sendUserError("Please log in", res);
+    return;
+  }
+  User.findById(sess.LoginId, (err, user) => {
+    if (err) {
+      sendUserError(err, res);
+      return;
+    }
+    req.user = user;
+  });
+});
+
 // TODO: implement routes
+server.post("/users", (req, res) => {
+  // getting the username and password from a user
+  const { username, password } = req.body;
+  // Check if the password is empty
+  if (password === "") {
+    sendUserError("Please input a valid password", res);
+    return;
+  }
+  // hash the password
+  bcrypt.hash(password, BCRYPT_COST, (err, passwordHash) => {
+    if (err) {
+      sendUserError(err, res);
+      return;
+    }
+    // const passwordHash = hPassword;
+    const newUser = new User({ username, passwordHash });
+    newUser.save((error, user) => {
+      if (error) {
+        sendUserError(error, res);
+        return;
+      }
+      res.json(user);
+    });
+  });
+});
+server.post("/login", (req, res) => {
+  const { username, password } = req.body;
+  const sess = req.session;
+  if (!password || password === "") {
+    sendUserError("Please input a valid password", res);
+  }
+  if (!username) {
+    sendUserError("Please input username", res);
+  }
+  // hash the password
+  User.findOne({ username }, (err, user) => {
+    if (err) {
+      sendUserError(err, res);
+      return;
+    }
+    if (!user) {
+      sendUserError("Please input valid info", res);
+      return;
+    }
+    if (bcrypt.compareSync(password, user.passwordHash)) {
+      sess.LoginId = user.id;
+      res.json({ success: true });
+      return;
```

Comment:
Whenever you call `res`, i.e. with `res.json` or `res.send`, there's an implicit `return`, so you don't need to call `return` here since you're already calling `res.json`